### DISCUSSION TITLE
Defining custom theme

### DISCUSSION THREAD

**Discussion by awa5114** [JanDeDobbeleer/oh-my-posh#1337] [Troubleshoot] [answered]
I'm trying to create a custom configuration to use with oh-my-posh. I already defined a `custom.omp.json` file based on two pre-existing themes. I would like to now attempt to make this configuration permanent for my terminal.

I tried the following to simply test out my configuration:

```
 oh-my-posh --config custom.omp.json --shell uni
```

However this did not work and I get an error:

> The term 'oh-my-posh' is not recognized as the name of a cmdlet, function, script file, or operable program.

I found an [issue](https://github.com/JanDeDobbeleer/oh-my-posh/issues/558)  from someone who has a similar problem, but I couldn't make much of it.

The other solution I can think of is placing my `custom.omp.json` file in the same directory as the themes included out-of-the-box, but I'm not sure if this is very good practice.

Would you please advise on how to proceed with this?

**Comment by lnu** [comment]
if you installed omp using powershell the executable is not available in the command prompt as is. The binary is located inside the powershell module. You can get the the binary location using ` (gmo -l oh-my-posh).path`.

**Reply by awa5114** [reply]
Thanks, but there are two issues:
 - `get-module` returns multiple results. Why does upgrading `oh-my-posh` simply install a new version on top of the old version as opposed to overwriting the older version?
```
C:\Users\921479\Documents\WindowsPowerShell\Modules\oh-my-posh\6.17.0\oh-my-posh.psd1
C:\Users\921479\Documents\WindowsPowerShell\Modules\oh-my-posh\5.6.4\oh-my-posh.psd1
```
 - Assuming the newer version is more applicable, there are multiple files with the name `oh-my-posh` inside that folder. Which one should I run?
```


    Directory: 
    C:\Users\921479\Documents\WindowsPowerShell\Modules\oh-my-posh\6.17.0


Mode                 LastWriteTime         Length Name                     
----                 -------------         ------ ----                     
d-----        27/11/2021     13:31                themes                   
-a----        26/11/2021     17:25            952 deploy.ps1               
-a----        26/11/2021     17:26           2157 oh-my-posh.psd1          
-a----        26/11/2021     17:25           8974 oh-my-posh.psm1          
-a----        26/11/2021     17:25            861 README.md                
```

**Reply by lnu** [reply]
for the multiple modules, that's how powershell modules work, it never removes the previous one. There was a change recently and now the latest version of the executable is always downloaded in the `oh-my-posh` folder which is for you `C:\Users\921479\Documents\WindowsPowerShell\Modules\oh-my-posh\oh-my-posh.exe`

=> better use set-poshprompt as @JanDeDobbeleer  said

**Comment by JanDeDobbeleer** [comment, marked as answer]
Don't overthink it and please read the docs. The [module docs](https://ohmyposh.dev/docs/pwsh#replace-your-existing-prompt) state you need to use `Set-PoshPrompt` to use another theme (use the full path, not the name of the file).

About the multiple versions, that's PowerShell, not us. Can't change that behavior.

**Reply by awa5114** [reply]
@JanDeDobbeleer I'm familiar with the documentation that you provide, but I'd overlooked its relevance in the current context. I now see what you mean. Adding- 1) `set-poshprompt` and 2) the absolute path to my custom theme in `$profile` makes it permanent.

I still believe hard-coding the absolute path like that is a little crude. I was hoping a more systematic way of handling (possibly multiple) custom themes existed. But that's just my personal opinion.

That's good to know about Powershell module versions. I had intended my comment to be more of a question rather than as criticism. Sorry if it came across like that.

**Reply by JanDeDobbeleer** [reply]
It didn't, written communication is hard (especially if people do in between everything else). Good to hear it works!

**Reply by JanDeDobbeleer** [reply]
Thinking about this whole conversation, I think I know what I can to do avoid confusion going forward. The module should be a means to install rather than a standalone thing.
